Improved Virtual Satellite Deployment

PhilMFischer opened the pull request:
- Now deploying executable binaries to sourceforge as well
- Added manuals to be deplyoed to GitHub Releases
- Cleaned up travis.yml and created additional bash scripts for build

Closes #125 .

Commit:
Trying to outsource deployment into bash script - (Task #125)

- Added new bash script that takes care of deploying
- Added deploy of zipped binaries for windows and linux as well
- Adjusted travis.yml accordingly

---
Task #125: Deploy Installable Packages to sourceforge as well

Commit:
Forcing failure to trigger upload - (Task #125)

- Forcing a failure in SWTBot tests to see if upload works as expected.

---
Task #125: Deploy Installable Packages to sourceforge as well

Commit:
Make this feature branch deploy to sourceforge - (Task #125)

- need to see if call to scripts are working.

---
Task #125: Deploy Installable Packages to sourceforge as well

Commit:
Switched to lower case letters for options - (Task #125)

- Lower case is used in all other places as well.

---
Task #125: Deploy Installable Packages to sourceforge as well

Commit:
Adding new shell build script to simplify travis - (Task #125)

- tarvis yml is growing to big. Try to reduce complexity by external
shells scripts

---
Task #125: Deploy Installable Packages to sourceforge as well

Commit:
Merge branch 'feature/task_125-Deploy_Installable_Pa' of https://github.com/virtualsatellite/VirtualSatellite4-Core into feature/task_125-Deploy_Installable_Pa

Commit:
adjusted file rights

Commit:
converted dos2unix

Commit:
Fixed printUsage call in scripts - (Task #125)

script had incorrect method name included

---
Task #125: Deploy Installable Packages to sourceforge as well

Commit:
fixed message in usage info - (Task #125)

- described wrong parameter

---
Task #125: Deploy Installable Packages to sourceforge as well

Commit:
Reduced Travis.yml file with external script - (Task #125)

- Added script for maven calls and builds
- calling bash maven build script from yaml

---
Task #125: Deploy Installable Packages to sourceforge as well

Commit:
Fixed maven build script - (Task #125)

- fixed function mavenCheckstyle which was not correctly spelled.
- improved usage help

---
Task #125: Deploy Installable Packages to sourceforge as well

Commit:
Fixing script call in travis.yml for deployment - (Task #125)

- Call in yaml to bash script not working yet. Trying to follow travis
example.

---
Task #125: Improve deployment and travis.yml

Commit:
Fixing test call to deploy - (Task #125)

- Try to deploy from feature branch before PR into development
- Attaching PDF document to GitHub Releases as well

---
Task #125: Improve deployment and travis.yml

Commit:
Fixing test call to deploy - (Task #125)

- Try to deploy from feature branch before PR into development

---
Task #125: Improve deployment and travis.yml

Commit:
Enabled to deploy on all branches for testing - (Task #125)

- Apparently travis only deploys from the default branch. Therefore we
now mention all branches.

---
Task #125: Improve deployment and travis.yml

Commit:
Removing test statements for deploy - (Task #125)

- getting script ready for PR

---
Task #125: Improve deployment and travis.yml

franzTobiasDLR commented on bash/maven_build.sh:
```diff
@@ -0,0 +1,142 @@
+	echo "Ant jacoco Reports"
+	ant jacocoPrepareDependencies
+	ant jacocoReport 2>&1 | tee ant.log
+	(grep -n "Rule violated" ant.log || exit 0 && exit 1;)
```
The build can also fail here if a e.g. a project does not have a source directory... Then surefire doesn't know how to handle the project. If that happens reports are not evaluated and the build succeeds without checking the coverage. "grep -n "Rule violated\|BUILD FAILED" ant.log" solves this. I have a fix for that on the requirements branch and can merge it later... Just to let you know ;)

PhilMFischer replied:
Good idea. I will include that suggestion.

franzTobiasDLR reviewed (dismissed):
Nice!

Commit:
Improved error handling for generating jacoco reports - (Task #125)

- Trying to grep "Build Failed" from ant log.

---
Task #125: Improve deployment and travis.yml

Commit:
Fixed damage in travis.yml - (Task #125)

- added incorrect content to file.

---
Task #125: Improve deployment and travis.yml

SaMuellerDLR reviewed (approved):
Approved!

PhilMFischer commented:
@franzTobiasDLR Please check your dismissed review.
